src/AstGen.zig

```diff
@@ -3394,6 +3396,16 @@ fn builtinCall(
             const result = try gz.addUnNode(.import, target, node);
             return rvalue(mod, scope, rl, result, node);
         },
+        .error_to_int => {
+            const target = try expr(mod, scope, .none, params[0]);
+            const result = try gz.addUnNode(.error_to_int, target, node);
+            return rvalue(mod, scope, rl, result, node);
+        },
+        .int_to_error => {
+            const target = try expr(mod, scope, .none, params[0]);
```

Comment:
Here pass a `.{.ty = .u16_type }` result location instead of `.none` and then actually remove the `coerce` call inside Sema

cc @Vexu